api/v1/referrals/views.py

```diff
@@ -18,22 +18,35 @@ def create_referral():
     email = phone = country = name = organization = None
     data = request.json or request.form
     data = dict(data)
-    base_lang = data.get('flow_base_language')[0]
-    contact = data.get('contact')[0]
-    values = json.loads(data.get('values')[0])
-    for v in values:
-        if v.get('category').get(base_lang).lower() in ['other', 'invalid']:
-            continue
-        if v.get('label').lower() == 'name':
-            name = v.get('text')
-        if v.get('label').lower() == 'phone number':
-            phone = v.get('rule_value')
-        if v.get('label').lower() == 'country':
-            country = v.get('text')
-        if v.get('label').lower() == 'email':
-            email = v.get('text')
-        if v.get('label').lower() == 'organization':
-            organization = v.get('category').get(base_lang).lower()
+
+    # check if we are a legacy webhook
+    if 'flow_base_language' in data:
+        base_lang = data.get('flow_base_language')[0]
+        contact = data.get('contact')[0]
+        values = json.loads(data.get('values')[0])
+        for v in values:
+            if v.get('category').get(base_lang).lower() in ['other', 'invalid']:
+                continue
+            if v.get('label').lower() == 'name':
+                name = v.get('text')
+            if v.get('label').lower() == 'phone number':
+                phone = v.get('rule_value')
+            if v.get('label').lower() == 'country':
+                country = v.get('text')
+            if v.get('label').lower() == 'email':
+                email = v.get('text')
+            if v.get('label').lower() == 'organization':
+                organization = v.get('category').get(base_lang).lower()
+    else:
+        # the new webhook format
+        contact = data.get('contact')['uuid']
+        results = data.get('results')
+        name = results.get('name')['input']
+        phone = results.get('phone_number')['input']
+        country = results.get('country')['input']
+        organization = results.get('organization')['category']
+        email = results.get('email')['input']
```

Comment:
One difference would be that the legacy code path doesn't blow up if any results are missing. Could use `results.get('email', {}).get('input')` to maintain that behavior.

Reply:
@ewheeler i see you've merged this but just want to highlight this in case it turns out to be an issue. I'm not sure how this webhook is used so not sure if it is.